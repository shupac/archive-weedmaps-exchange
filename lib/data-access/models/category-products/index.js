import { types } from 'mobx-state-tree';
import {
  BuyerProduct,
  type ProductCardType,
} from 'lib/data-access/models/product';

// TODO: update products with products array types as data is available
const CategoryProducts = types.model('CategoryProducts', {
  slug: types.string,
  position: types.number,
  name: types.string,
  id: types.string,
  buyerProducts: types.array(BuyerProduct),
});

export type CategoryProductsType = {
  id: string,
  name: string,
  slug: string,
  position: number,
  products: ProductCardType[],
};

export default CategoryProducts;
