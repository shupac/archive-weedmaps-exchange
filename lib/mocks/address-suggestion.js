export const mockSuggestedAddresses = [
  {address: "41 Discovery Park Boulevard, Seattle, WA, USA", id: "Ei00MSBEaXNjb3ZlcnkgUGFyayBCb3VsZXZhcmQsIFNlYXR0bGUsIFdBLCBVU0E"},
  {address: "41 Discovery Drive, The Dalles, OR, USA", id: "Eic0MSBEaXNjb3ZlcnkgRHJpdmUsIFRoZSBEYWxsZXMsIE9SLCBVU0E"},
  {address: "41 Discovery Circle Northeast, Issaquah, WA, USA", id: "EjA0MSBEaXNjb3ZlcnkgQ2lyY2xlIE5vcnRoZWFzdCwgSXNzYXF1YWgsIFdBLCBVU0E"},
  {address: "41 Discovery Trail, Vancouver, WA, USA", id: "EiY0MSBEaXNjb3ZlcnkgVHJhaWwsIFZhbmNvdXZlciwgV0EsIFVTQQ"},
  {address: "41 Discovery Street, Walla Walla, WA, USA", id: "Eik0MSBEaXNjb3ZlcnkgU3RyZWV0LCBXYWxsYSBXYWxsYSwgV0EsIFVTQQ"},
]
