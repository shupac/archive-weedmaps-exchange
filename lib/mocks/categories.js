
export const mockCategories = [
  {
    "slug":"concentrates",
    "position":1,
    "name":"Concentrates",
    "id":"81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/concentrates.svg",
      "id":"c77396fe-4293-4911-805c-2cd3afb18949"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/concentrates.png",
      "id":"82c8a744-10c6-48c9-b5a0-3a90de6d2c2f"
    }
  },
  {
    "slug":"edibles",
    "position":3,
    "name":"Edibles",
    "id":"e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "iconImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/icons/edibles.svg",
      "id":"b252a9f9-f06e-4981-80db-cf74d6c7573a"
    },
    "avatarImage":{
      "smallUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "mediumUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "largeUrl":"https://wm-exchange-assets-acceptance.s3.amazonaws.com/categories/images/edibles.png",
      "id":"97cc29eb-0ce4-4b55-81ae-e1899b6dc7ce"
    }
  }
];