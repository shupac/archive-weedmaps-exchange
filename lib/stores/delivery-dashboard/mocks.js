export const mockIndex = {
  data: {
    id: '1',
    type: 'index_card',
    relationships: {
      driver_cards: {
        data: [
          {
            type: 'driver_card',
            id: '1',
          },
          {
            type: 'driver_card',
            id: '2',
          },
          {
            type: 'driver_card',
            id: '3',
          },
        ],
      },
      route_cards: {
        data: [
          {
            type: 'route_card',
            id: '1',
          },
        ],
      },
      delivery_cards: {
        data: [
          {
            type: 'delivery_card',
            id: '1',
          },
          {
            type: 'delivery_card',
            id: '2',
          },
        ],
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
  included: [
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'started',
        destination_address: '1234',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Alan Tirado',
        contact_email: 'atirado@weedmaps.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '123-123-1234',
        item_count: 4,
        order_id: '1234',
        total: 1,
      },
      id: '1',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'created',
        destination_address: '11234 W S St',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Bob Truman',
        contact_email: 'bob@truman.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '343-345-3452',
        item_count: 4,
        order_id: '3421',
        total: 3455,
      },
      id: '2',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'started',
        destination_address: '11234 W S St',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Bob Truman',
        contact_email: 'bob@truman.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '343-345-3452',
        item_count: 4,
        order_id: '3421',
        total: 3455,
      },
      id: '3',
      relationships: {
        package: {
          data: {
            id: '3',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'assigned',
      },
      id: '1',
      relationships: {
        delivery_cards: {
          data: [
            {
              id: '1',
              type: 'delivery_card',
            },
          ],
        },
        driver: {
          data: {
            id: '1',
            type: 'driver_card',
          },
        },
      },
      type: 'route_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'in_progress',
      },
      id: '2',
      relationships: {
        delivery_cards: {
          data: [
            {
              id: '3',
              type: 'delivery_card',
            },
          ],
        },
        driver: {
          data: {
            id: '2',
            type: 'driver_card',
          },
        },
      },
      type: 'route_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'on_duty',
        name: 'Bob',
        phone_number: '714-277-9631',
        avatar_url:
          'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
      },
      id: '1',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [
            {
              id: '1',
              type: 'route_card',
            },
          ],
        },
      },
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'on_duty',
        name: 'James',
        phone_number: '123-123-1234',
        avatar_url:
          'https://images.weedmaps.com/users/000/331/653/avatar/square_fill/1510567490-ShortRound.jpg?development=1',
      },
      id: '2',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'off_duty',
        name: 'Val',
        phone_number: '666-666-5656',
        avatar_url: null,
      },
      id: '3',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
  ],
};
export const mockBaseIndex = {
  data: [
    {
      id: '1',
      type: 'index_card',
      relationships: {
        driver_cards: {
          data: [
            {
              type: 'driver_card',
              id: '1',
            },
          ],
        },
        route_cards: {
          data: [],
        },
        delivery_cards: {
          data: [
            {
              type: 'delivery_card',
              id: '1',
            },
          ],
        },
      },
    },
  ],
  jsonapi: {
    version: '1.0',
  },
  included: [
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'successful',
        destination_address: '1234',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'String',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: 'String',
        item_count: 4,
        order_id: 'String',
        total: 1,
      },
      id: '1',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'free',
        name: 'Bob',
        phone_number: '714-277-9631',
        avatar_url:
          'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
      },
      id: '1',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
  ],
};
export const mockNewRoute = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'assigned',
    },
    id: '1',
    relationships: {
      delivery_cards: {
        data: [
          {
            id: '1',
            type: 'delivery_card',
          },
        ],
      },
      driver: {
        data: {
          id: '1',
          type: 'driver_card',
        },
      },
    },
  },
  included: [],
  jsonapi: { version: '1.0' },
};
export const mockNewDelivery = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'created',
      destination_address: '41 Discovery',
      destination_latitude: 33.666629,
      destination_longitude: -117.756304,
      recipient_name: 'Alan Tirado',
      ordered_at: '2018-05-08T23:00:42.075253Z',
      contact_number: '123-123-1234',
      item_count: 4,
      order_id: '420',
      total: 420,
    },
    id: '2',
    relationships: {
      package: {
        data: {
          id: '2',
          type: 'package',
        },
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockNewDriver = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'off_duty',
      name: 'Bob',
      phone_number: '123-123-1234',
      avatar_url:
        'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
    },
    id: '3',
    type: 'driver_card',
    relationships: {
      in_progress_route_cards: {
        data: [],
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockDriverLocation = {
  data: {
    id: '1',
    type: 'driver_location',
    attributes: {
      latitude: 33.666629,
      longitude: -117.756304,
      occurred_at: '2018-05-08T23:00:42.075253Z',
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockPackage = {
  data: {
    attributes: {
      contact_email: 'labron@weedmaps.com',
      contact_number: '555-555-5555',
      documents: [
        {
          image_url:
            'https://images-acceptance.internal-weedmaps.com/users/000/331/653/avatar/square_fill/1510567490-ShortRound.jpg',
          name: 'id',
        },
        {
          image_url:
            'https://images-acceptance.internal-weedmaps.com/users/002/306/378/avatar/square_fill/1531352592-download.jpeg',
          name: 'rec',
        },
      ],
      excise_tax: 1,
      inserted_at: '2018-05-08T23:00:42.075253Z',
      order_id: 'UUID-1234',
      order_items: [
        {
          name: 'Hardcore Og',
          price: 50,
          quantity: 1,
        },
      ],
      ordered_at: '2018-05-08T23:00:42.075253Z',
      recipient_name: 'Bob',
      recipient_note: 'Quick yo!',
      sales_tax: 1,
      subtotal: 1,
      total: 1,
      updated_at: '2018-05-08T23:00:42.075253Z',
    },
    id: '1234',
    type: 'package',
  },
  jsonapi: {
    version: '1.0',
  },
};
