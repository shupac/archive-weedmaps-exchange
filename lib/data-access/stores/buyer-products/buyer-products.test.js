import { getSnapshot } from 'mobx-state-tree';
import { mockProducts } from 'lib/mocks/search-results';
import { mockCategoryProducts } from 'lib/mocks/category-products';
import BuyerProducts, { regionId } from './';

describe('BuyerProducts Store', () => {
  it('can fetch category products and set data', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockCategoryProducts),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.getCategoryProducts({ search: 'indica' });
    expect(mockFetchClient.fetch).toHaveBeenCalledWith(
      `/buyer/regions/${regionId}/departments?include=products`,
    );
    expect(getSnapshot(buyerProducts.categoryProductsData)).toMatchSnapshot();
  });

  it('can fetch product search and set data', () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    expect(buyerProducts.searchResultsLoading).toEqual(false);
    const promise = buyerProducts.searchCatalog({ search: 'indica' });
    expect(mockFetchClient.fetch).toHaveBeenCalledWith(
      `/buyer/regions/${regionId}/products?query=indica`,
    );
    expect(buyerProducts.searchResultsLoading).toEqual(true);
    promise.then(() => {
      expect(getSnapshot(buyerProducts.searchResultsData)).toMatchSnapshot();
      expect(buyerProducts.searchResultsLoading).toEqual(false);
    });
  });

  it('should not fetch product search with no query', () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    buyerProducts.searchCatalog({});
    expect(mockFetchClient.fetch).not.toHaveBeenCalled();
    expect(buyerProducts.searchResults).toEqual(null);
  });

  it('can fetch product details and set data', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts[0]),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.getProductDetails('1234');
    expect(mockFetchClient.fetch).toHaveBeenCalledWith(
      `/buyer/regions/${regionId}/products/1234`,
    );
    expect(getSnapshot(buyerProducts.productDetailsData)).toMatchSnapshot();
  });

  it('can return the category products view', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockCategoryProducts),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.getCategoryProducts({ search: 'indica' });
    expect(buyerProducts.categoryProducts).toMatchSnapshot();
  });

  it('can return the search results view', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.searchCatalog({ search: 'indica' });
    expect(buyerProducts.searchResults).toMatchSnapshot();
  });

  it('should return null for no search results', () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    buyerProducts.setSearchResultsData([]);
    expect(buyerProducts.searchResults).toEqual(null);
  });

  it('can return the product details view', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts[0]),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.getProductDetails('1234');
    expect(buyerProducts.productDetails).toMatchSnapshot();
  });

  it('can return the product variants view', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts[0]),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.getProductDetails('1234');
    expect(buyerProducts.productVariants).toMatchSnapshot();
  });

  it('can return the product breadcrumb view', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(mockProducts[0]),
    };
    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockFetchClient,
      },
    );
    await buyerProducts.getProductDetails('1234');
    expect(buyerProducts.productBreadcrumb).toMatchSnapshot();
  });

  it('should catch errors', () => {
    const mockErrorClient = {
      fetch: () => {
        throw new Error('test');
      },
    };

    const log = jest.spyOn(global.console, 'log').mockReturnValue();

    const buyerProducts = BuyerProducts.create(
      {},
      {
        client: mockErrorClient,
      },
    );
    buyerProducts.getCategoryProducts();
    expect(log).toHaveBeenCalledWith(new Error('test'));
    buyerProducts.searchCatalog({ search: 'foo' });
    expect(log).toHaveBeenCalledWith(new Error('test'));
    buyerProducts.getProductDetails('foo');
    expect(log).toHaveBeenCalledWith(new Error('test'));

    log.mockRestore();
  });
});
