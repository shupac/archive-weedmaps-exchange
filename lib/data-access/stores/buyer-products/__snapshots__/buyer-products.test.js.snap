// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyerProducts Store can fetch category products and set data 1`] = `
Array [
  Object {
    "buyerProducts": Array [],
    "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "name": "Concentrates",
    "position": 1,
    "slug": "concentrates",
  },
  Object {
    "buyerProducts": Array [],
    "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "name": "Edibles",
    "position": 3,
    "slug": "edibles",
  },
]
`;

exports[`BuyerProducts Store can fetch product details and set data 1`] = `
Object {
  "id": "5cd4463d-52bf-4b18-84f4-e3b378fbc989",
  "priceRanges": Array [
    Object {
      "maxPrice": 20,
      "minPrice": 10,
      "unit": "gram",
    },
  ],
  "product": Object {
    "avatarImage": Object {
      "id": "41c06b6d-099c-48c4-82c6-015d3b8dcc3e",
      "largeUrl": "https://images.weedmaps.com/products/000/023/221/avatar/large/gfarmsalab_nug-run-indica_2.jpg?development=1",
      "mediumUrl": "https://images.weedmaps.com/products/000/023/221/avatar/medium/gfarmsalab_nug-run-indica_2.jpg?development=1",
      "smallUrl": "https://images.weedmaps.com/products/000/023/221/avatar/square/gfarmsalab_nug-run-indica_2.jpg?development=1",
    },
    "brand": Object {
      "avatarImage": undefined,
      "deliveryEta": undefined,
      "description": "<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Leader of the American Cannabis Revolution</em></p> ",
      "id": "d054b34f-428e-4583-a2a4-27f4e5aea6f3",
      "licenses": Array [
        Object {
          "id": "8c4369fc-a767-4e5c-9863-b5225705d563",
          "licenseType": "Sales",
          "number": "420710",
        },
      ],
      "minimumPurchasePrice": 123,
      "name": "GfarmaLabs",
      "shippingFee": undefined,
      "slug": "gfarmalabs",
    },
    "categories": Array [],
    "departments": Array [
      Object {
        "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
        "name": "Concentrates",
        "slug": "concentrates",
      },
    ],
    "description": "<p>G FarmaLabs is proud to bring you NUG RUN Cannabis Oil Injectors. This pharmaceutical grade Co2 extract is created by using 100% IndoorCannabis Flower and then further removing any undesirable compounds. By purifying the essential oils of what remains, Liquid Gold™ NUG RUN was born. Liquid Gold™ NUG RUN Cannabis Oil was developed to better serve Patients through the form of vaporizing. Cannabis vaporizers are designed to let users inhale active Cannabinoids while avoiding harmful smoke toxins. They do so by heating cannabis to a temperature of 180 - 200° C (356° - 392°), Just below the point of combustion where smoke is produced. By vaporizing, THC and other medically active Cannabinoids are emitted with little to none of the carcinogenic tars and noxious gasses found in smoke. Many medical marijuana patients who find smoking marijuana highly irritating, report effective relief through vaporizers. Users who are concerned about the respiratory hazards of smoking are strongly advised to use vaporizers. Liquid Gold™ NUG RUN Cannabis Oil Injectors is available in .5 GRAM Injector, manufactured to fit most Vape Pen batteries in the marketplace.</p>",
    "galleryImages": Array [
      Object {
        "id": "1608ec15-013e-4a77-9cb9-27cc232a6640",
        "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
        "mediumUrl": "https://images.weedmaps.com/photos/products/000/023/221/medium/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
        "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
      },
      Object {
        "id": "b2d1cf18-e1e1-443a-985a-bec6b104eaac",
        "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
        "mediumUrl": "https://images.weedmaps.com/photos/products/000/023/221/medium/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
        "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
      },
    ],
    "id": "5cd4463d-52bf-4b18-84f4-e3b378fbc989",
    "name": "NUG RUN Injector, Indica",
    "slug": "gfarmalabs-nug-run-injector-indica",
    "variants": Array [
      Object {
        "allocations": Array [
          Object {
            "active": true,
            "amount": 81,
            "currency": "usd",
            "id": "091ed653-2f28-4904-89b8-5ba31330a21d",
            "price": "30",
            "zone": undefined,
          },
        ],
        "id": "5cfce13e-b8b9-4da0-a363-4becc6a06bb9",
        "name": "Eighth Pack",
        "size": 3.5,
        "sku": "12345",
        "unit": "gram",
      },
    ],
  },
}
`;

exports[`BuyerProducts Store can fetch product details and set data 2`] = `
Object {
  "id": "1608ec15-013e-4a77-9cb9-27cc232a6640",
  "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
  "mediumUrl": "https://images.weedmaps.com/photos/products/000/023/221/medium/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
  "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
}
`;

exports[`BuyerProducts Store can fetch product search and set data 1`] = `
Array [
  Object {
    "id": "5cd4463d-52bf-4b18-84f4-e3b378fbc989",
    "priceRanges": Array [
      Object {
        "maxPrice": 20,
        "minPrice": 10,
        "unit": "gram",
      },
    ],
    "product": Object {
      "avatarImage": Object {
        "id": "41c06b6d-099c-48c4-82c6-015d3b8dcc3e",
        "largeUrl": "https://images.weedmaps.com/products/000/023/221/avatar/large/gfarmsalab_nug-run-indica_2.jpg?development=1",
        "mediumUrl": "https://images.weedmaps.com/products/000/023/221/avatar/medium/gfarmsalab_nug-run-indica_2.jpg?development=1",
        "smallUrl": "https://images.weedmaps.com/products/000/023/221/avatar/square/gfarmsalab_nug-run-indica_2.jpg?development=1",
      },
      "brand": Object {
        "avatarImage": undefined,
        "deliveryEta": undefined,
        "description": "<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Leader of the American Cannabis Revolution</em></p> ",
        "id": "d054b34f-428e-4583-a2a4-27f4e5aea6f3",
        "licenses": Array [
          Object {
            "id": "8c4369fc-a767-4e5c-9863-b5225705d563",
            "licenseType": "Sales",
            "number": "420710",
          },
        ],
        "minimumPurchasePrice": 123,
        "name": "GfarmaLabs",
        "shippingFee": undefined,
        "slug": "gfarmalabs",
      },
      "categories": Array [],
      "departments": Array [
        Object {
          "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
          "name": "Concentrates",
          "slug": "concentrates",
        },
      ],
      "description": "<p>G FarmaLabs is proud to bring you NUG RUN Cannabis Oil Injectors. This pharmaceutical grade Co2 extract is created by using 100% IndoorCannabis Flower and then further removing any undesirable compounds. By purifying the essential oils of what remains, Liquid Gold™ NUG RUN was born. Liquid Gold™ NUG RUN Cannabis Oil was developed to better serve Patients through the form of vaporizing. Cannabis vaporizers are designed to let users inhale active Cannabinoids while avoiding harmful smoke toxins. They do so by heating cannabis to a temperature of 180 - 200° C (356° - 392°), Just below the point of combustion where smoke is produced. By vaporizing, THC and other medically active Cannabinoids are emitted with little to none of the carcinogenic tars and noxious gasses found in smoke. Many medical marijuana patients who find smoking marijuana highly irritating, report effective relief through vaporizers. Users who are concerned about the respiratory hazards of smoking are strongly advised to use vaporizers. Liquid Gold™ NUG RUN Cannabis Oil Injectors is available in .5 GRAM Injector, manufactured to fit most Vape Pen batteries in the marketplace.</p>",
      "galleryImages": Array [
        Object {
          "id": "1608ec15-013e-4a77-9cb9-27cc232a6640",
          "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
          "mediumUrl": "https://images.weedmaps.com/photos/products/000/023/221/medium/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
          "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
        },
        Object {
          "id": "b2d1cf18-e1e1-443a-985a-bec6b104eaac",
          "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
          "mediumUrl": "https://images.weedmaps.com/photos/products/000/023/221/medium/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
          "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
        },
      ],
      "id": "5cd4463d-52bf-4b18-84f4-e3b378fbc989",
      "name": "NUG RUN Injector, Indica",
      "slug": "gfarmalabs-nug-run-injector-indica",
      "variants": Array [
        Object {
          "allocations": Array [
            Object {
              "active": true,
              "amount": 81,
              "currency": "usd",
              "id": "091ed653-2f28-4904-89b8-5ba31330a21d",
              "price": "30",
              "zone": undefined,
            },
          ],
          "id": "5cfce13e-b8b9-4da0-a363-4becc6a06bb9",
          "name": "Eighth Pack",
          "size": 3.5,
          "sku": "12345",
          "unit": "gram",
        },
      ],
    },
  },
]
`;

exports[`BuyerProducts Store can return the category products view 1`] = `
Array [
  Object {
    "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "name": "Concentrates",
    "products": Array [],
  },
  Object {
    "id": "e306a27d-5076-4074-ab7a-21f6485ad4bb",
    "name": "Edibles",
    "products": Array [],
  },
]
`;

exports[`BuyerProducts Store can return the product breadcrumb view 1`] = `
Array [
  Object {
    "id": "81ac71b1-1c0c-490e-ad8a-062a6a1ae9a6",
    "name": "Concentrates",
    "slug": "concentrates",
  },
]
`;

exports[`BuyerProducts Store can return the product details view 1`] = `
Object {
  "brand": "GfarmaLabs",
  "description": "<p>G FarmaLabs is proud to bring you NUG RUN Cannabis Oil Injectors. This pharmaceutical grade Co2 extract is created by using 100% IndoorCannabis Flower and then further removing any undesirable compounds. By purifying the essential oils of what remains, Liquid Gold™ NUG RUN was born. Liquid Gold™ NUG RUN Cannabis Oil was developed to better serve Patients through the form of vaporizing. Cannabis vaporizers are designed to let users inhale active Cannabinoids while avoiding harmful smoke toxins. They do so by heating cannabis to a temperature of 180 - 200° C (356° - 392°), Just below the point of combustion where smoke is produced. By vaporizing, THC and other medically active Cannabinoids are emitted with little to none of the carcinogenic tars and noxious gasses found in smoke. Many medical marijuana patients who find smoking marijuana highly irritating, report effective relief through vaporizers. Users who are concerned about the respiratory hazards of smoking are strongly advised to use vaporizers. Liquid Gold™ NUG RUN Cannabis Oil Injectors is available in .5 GRAM Injector, manufactured to fit most Vape Pen batteries in the marketplace.</p>",
  "galleryImages": Array [
    Object {
      "id": "1608ec15-013e-4a77-9cb9-27cc232a6640",
      "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
      "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29923_gfarmsalab_nug-run-indica_2.jpg?development=1",
    },
    Object {
      "id": "b2d1cf18-e1e1-443a-985a-bec6b104eaac",
      "largeUrl": "https://images.weedmaps.com/photos/products/000/023/221/large/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
      "smallUrl": "https://images.weedmaps.com/photos/products/000/023/221/square/29969_gfarmsalab_nug-run-indica_1.jpg?development=1",
    },
  ],
  "id": "5cd4463d-52bf-4b18-84f4-e3b378fbc989",
  "licenses": Array [
    Object {
      "id": "8c4369fc-a767-4e5c-9863-b5225705d563",
      "licenseType": "Sales",
      "number": "420710",
    },
  ],
  "name": "NUG RUN Injector, Indica",
  "priceRanges": Array [
    Object {
      "maxPrice": 20,
      "minPrice": 10,
      "unit": "gram",
    },
  ],
}
`;

exports[`BuyerProducts Store can return the product variants view 1`] = `
Array [
  Object {
    "amount": 81,
    "id": "5cfce13e-b8b9-4da0-a363-4becc6a06bb9",
    "inStock": true,
    "name": "Eighth Pack",
    "price": 30,
    "size": 3.5,
    "unit": "gram",
  },
]
`;

exports[`BuyerProducts Store can return the search results view 1`] = `
Array [
  Object {
    "brand": "GfarmaLabs",
    "category": "Concentrates",
    "id": "5cd4463d-52bf-4b18-84f4-e3b378fbc989",
    "imageUrl": "https://images.weedmaps.com/products/000/023/221/avatar/medium/gfarmsalab_nug-run-indica_2.jpg?development=1",
    "name": "NUG RUN Injector, Indica",
    "outOfStock": false,
    "priceRanges": Array [
      Object {
        "maxPrice": 20,
        "minPrice": 10,
        "unit": "gram",
      },
    ],
  },
]
`;
